src/Symfony/Component/Lock/Store/MongoDbStore.php

```diff
@@ -142,6 +149,10 @@ public function __construct(Collection|Client|string $mongo, array $options = []
      */
     private function skimUri(string $uri): string
```

Comment:
Leaving this here: note that as per the [Connection Strings documentation](https://www.mongodb.com/docs/manual/reference/connection-string/), the path in the connection string refers to the default authentication database to be used if the `authSource` does not define one. While most people consider this to be the default database, I wanted to point out the difference. Not necessarily an issue of this PR as it's already the current behaviour, so feel free to resolve this comment after reading if you think this isn't worth changing.

Reply:
Sure, that would be better if the option `database` had precedence over the database in the path of the URI.